src/main/java/com/redhat/devtools/intellij/quarkus/module/QuarkusCodeEndpointChooserStep.java

```diff
@@ -115,34 +147,117 @@ private void updateCustomUrl() {
         this.endpointURL.setEnabled(custom);
         this.customUrlWithBrowseButton.setButtonEnabled(custom);
+        if (loadingRequest != null) {
+            loadingRequest.cancel(true);
+        }
+        loadingRequest = loadStreams();
+    }
+
+    private Future<QuarkusModel> loadStreams() {
+        final String endpoint = getSelectedEndpointUrl();
+        if (endpoint == null || endpoint.isBlank()) {
+            return null;
+        }
+        showSpinner();
+        return ApplicationManager.getApplication().executeOnPooledThread(() -> {
+            try {
+                return QuarkusModelRegistry.INSTANCE.loadStreams(endpoint, new EmptyProgressIndicator());
+            } catch (Exception e) {
+                ApplicationManager.getApplication().invokeLater(() -> {
+                    Messages.showErrorDialog("Failed to load Quarkus platform versions from "+endpoint+":\n"+ e.getMessage(), "Error loading Quarkus platform versions");
+                },  ModalityState.stateForComponent(getComponent()) );
+            } finally {
+                ApplicationManager.getApplication().invokeLater(() -> {
+                    hideSpinner();
+                }, ModalityState.stateForComponent(getComponent()));
+            }
+            return null;
+        });
+    }
+
+    private void showSpinner() {
+        spinner.setVisible(true);
+        spinner.resume();
+    }
+
+    private void hideSpinner() {
+        spinner.suspend();
+        spinner.setVisible(false);
     }
 
+    @Override
     public boolean validate() throws ConfigurationException {
-        if (this.defaultRadioButton.isSelected()) {
-            return true;
-        } else {
+        if (this.customRadioButton.isSelected()) {
             String serviceUrl = this.endpointURL.getText();
             if (serviceUrl.isEmpty()) {
                 throw new ConfigurationException("Quarkus Code endpoint URL must be set");
-            } else if (!serviceUrl.startsWith("http://") && !serviceUrl.startsWith("https://")) {
+            } else if (!serviceUrl.matches("^https?://.*")) {
                 throw new ConfigurationException("Invalid custom Quarkus Code endpoint URL");
-            } else {
-                try {
-                    QuarkusModelRegistry.INSTANCE.load(serviceUrl, new EmptyProgressIndicator());
-                    return true;
-                } catch (IOException var3) {
-                    throw new ConfigurationException("Invalid Custom Quarkus Code endpoint URL");
-                }
             }
         }
+        try {
+            boolean requestComplete = checkRequestComplete();
+            if (!requestComplete) {
+                return false;
+            }
+            streams = loadingRequest.get(DEFAULT_TIMEOUT_IN_SEC, TimeUnit.SECONDS);
+            if (streams == null) {
+                throw new ConfigurationException("Failed to load Quarkus platform streams");
+            }
+        } catch (Exception e) {
```

Comment:
Perhaps it should be better to distinguate timeout error and another errors in the message. I suggest that you catch TimeOutException and you defines a proper message for timeout.

Reply:
done